core/include/gnuradio-4.0/Block.hpp

```diff
@@ -1477,44 +1653,57 @@ class Block : public lifecycle::StateMachine<Derived>, //
     }
 
     void
-    processMessages(MsgPortInNamed<"__Builtin"> &port, std::span<const Message> messages) {
-        if (std::addressof(port) != std::addressof(msgIn)) {
-            fmt::print("{} got a message on a wrong port\n", self().unique_name);
-            return;
-        }
+    processMessages([[maybe_unused]] const MsgPortInNamed<"__Builtin"> &port, std::span<const Message> messages) {
+        using enum gr::message::Command;
+        assert(std::addressof(port) == std::addressof(msgIn) && "got a message on wrong port");
 
         for (const auto &message : messages) {
-            const auto kind   = messageField<std::string>(message, gr::message::key::Kind).value_or(std::string{});
-            const auto target = messageField<std::string>(message, gr::message::key::Target);
-
-            if (target && !target->empty() && *target != self().unique_name) {
+            if (!message.serviceName.empty() && message.serviceName != unique_name && message.serviceName != name) {
+                // Skip if target does not match the block's (unique) name and is not empty.
                 continue;
             }
 
-            if (kind == gr::message::kind::UpdateSettings) {
-                const auto data   = messageField<property_map>(message, gr::message::key::Data).value();
-                auto       notSet = settings().set(data);
-
-                std::string keysNotSet;
-                for (const auto &[k, v] : notSet) {
-                    keysNotSet += " " + k;
+            PropertyCallback callback = nullptr;
+            // Attempt to find a matching property callback or use the unmatchedPropertyHandler.
+            if (auto it = propertyCallbacks.find(message.endpoint); it != propertyCallbacks.end()) {
+                callback = it->second;
+            } else {
+                if constexpr (requires(std::string_view sv, Message m) {
+                                  { self().unmatchedPropertyHandler(sv, m) } -> std::same_as<std::optional<Message>>;
+                              }) {
+                    callback = &Derived::unmatchedPropertyHandler;
                 }
+            }
 
-                Message settingsUpdated;
-                settingsUpdated[gr::message::key::Kind] = gr::message::kind::SettingsChangeRequested;
-                settingsUpdated[gr::message::key::Data] = settings().get();
+            if (callback == nullptr) {
```

Comment:
callback is also nullptr when cmd::Set is processed, probably one can also add this to the docs.

Reply:
The logic of that part of the code has changed: the 'callback' points to either the registered function for a given property or the 'unmatchedPropertyHandler(...)' that should be invoked. This is before any MDP command is being queried/executed.

Reply:
Maybe the removed code was misleading... refactored this late because I found my original lambda-based logic needlessly complex.